src/hotspot/share/gc/shared/cardTable.cpp

```diff
@@ -262,3 +274,220 @@ void CardTable::print_on(outputStream* st) const {
+
+    bool res = os::uncommit_memory((char*)delta.start(),
+                                   delta.byte_size());
+    assert(res, "uncommit should succeed");
+  } else {
+    log_trace(gc, barrier)("Committed card table region unchanged");
+  }
+
+  MemRegion prev_committed_tenured = committed_for_region_in_shared_virtual_space(_covered[old_gen_idx]);
+  MemRegion committed_tenured = committed_for_region_in_shared_virtual_space(new_region0);
+  MemRegion committed_young = committed_for_region_in_shared_virtual_space(new_region1);
+
+  _covered[old_gen_idx] = new_region0;
+  _covered[young_gen_idx] = new_region1;
+
+#ifdef ASSERT
+  log_trace(gc, barrier)("CardTable::resize_covered_region_shared_virtual_space: ");
+  log_trace(gc, barrier)("    prev_committed.start():              " PTR_FORMAT "  prev_committed.last():              " PTR_FORMAT,
+                         p2i(prev_committed.start()), p2i(prev_committed.last()));
+  log_trace(gc, barrier)("    new_committed_start:                 " PTR_FORMAT "  new_committed_last:                 " PTR_FORMAT,
+                         p2i(new_committed.start()), p2i(new_committed.last()));
+  log_trace(gc, barrier)("    committed_tenured.start():           " PTR_FORMAT "  committed_tenured.last():           " PTR_FORMAT,
+                         p2i(committed_tenured.start()), p2i(committed_tenured.last()));
+  log_trace(gc, barrier)("    committed_young.start():             " PTR_FORMAT "  committed_young.last():             " PTR_FORMAT,
+                         p2i(committed_young.start()), p2i(committed_young.last()));
+  log_trace(gc, barrier)("    addr_for(committed_tenured.start()): " PTR_FORMAT "  addr_for(committed_tenured.last()): " PTR_FORMAT,
+                         p2i(addr_for((CardValue*) committed_tenured.start())),  p2i(addr_for((CardValue*) committed_tenured.last())));
+  log_trace(gc, barrier)("    addr_for(committed_young.start()):   " PTR_FORMAT "  addr_for(committed_young.last()):   " PTR_FORMAT,
+                         p2i(addr_for((CardValue*) committed_young.start())),  p2i(addr_for((CardValue*) committed_young.last())));
+
+  for (int idx=0; idx < 2; idx++) {
+    log_trace(gc, barrier)("    _covered[%d].start():           " PTR_FORMAT "  _covered[%d].last(): " PTR_FORMAT,
+                            idx, p2i(_covered[idx].start()),
+                            idx, p2i(_covered[idx].last()));
+    log_trace(gc, barrier)("    byte_for(_covered[%d].start()): " PTR_FORMAT "  byte_for(_covered[%d].last()): " PTR_FORMAT,
+                           idx, p2i(byte_for(_covered[idx].start())),
+                           idx, p2i(byte_for(_covered[idx].last())));
+  }
+#endif
+
+  assert(committed_tenured.last() < committed_young.start(), "last word of tenured must be less than first word of young gen");
+  assert(committed_young.last() <= new_committed.last(), "last word of young gen must be in committed card table memory");
```

Comment:
For historical reasons, are we destined to suffer the mixed use of "new" and "young" in GC code? I wish we could pick one and stick with it. It's also not clear to me why these variables have the "new"/"young" and "committed" in reversed order. Why does that have to be inconsistent? From someone just looking at this code for the first time, there's nothing that really distinguishes these two vars (and what they refer to) other than superficial conventions. It feels like something could be done to make things clearer.

Reply:
> For historical reasons, are we destined to suffer the mixed use of "new" and "young" in GC code? I wish we could pick one and stick with it. It's also not clear to me why these variables have the "new"/"young" and "committed" in reversed order. Why does that have to be inconsistent? From someone just looking at this code for the first time, there's nothing that really distinguishes these two vars (and what they refer to) other than superficial conventions. It feels like something could be done to make things clearer.

Part of the confusion here is that there are new values coming in/being computed - they are new when contrasted with the existing values they will replace. I have renamed the variables to clarify what they are referring to.